src/App.js

```diff
@@ -1,19 +1,42 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
 import chatMessages from './data/messages.json';
 
 const App = () => {
+  const [chatData, setChatData] = useState(chatMessages);
+
+  const updateChatData = (updatedChatData) => {
+    setChatData(
```

Comment:
Nice use of the callback style of setter, since our next state for the chat data depends on the previous state of the chat data.